cmake/modules/BuildFIO.cmake

```diff
@@ -0,0 +1,15 @@
+function(build_fio)
+  # we use an external project and copy the sources to bin directory to ensure
+  # that object files are built outside of the source tree.
+  include(ExternalProject)
+  ExternalProject_Add(fio_ext
+    DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+    GIT_REPOSITORY "https://github.com/axboe/fio.git"
+    GIT_TAG "e09d68c8da4ab91397490577454de928106651f5"
+    SOURCE_DIR ${CMAKE_BINARY_DIR}/src/fio
+    BINARY_DIR ${CMAKE_BINARY_DIR}/src/fio
+    CONFIGURE_COMMAND ${CMAKE_BINARY_DIR}/src/fio/configure
```

Comment:
nit, could use `<SOURCE_DIR>` as a token to reference the defined `SOURCE_DIR`, like 
```cmake
CONFIGURE_COMMAND <SOURCE_DIR>/configure
```